### DISCUSSION TITLE
findMany IN query with combination of values

### DISCUSSION THREAD

**Discussion by andreasvh-conceto** [prisma/prisma#17379] [Q&A] [answered]
Hello,

assuming i have the following table:
```
CREATE TABLE person (
    id SERIAL,
    last_name varchar(255),
    first_name varchar(255),
    city varchar(255)
);
```
and the related prisma model:
```
model person{
  id      Int      @id @default(autoincrement())
  last_name String
  first_name String
  city String
}
```
Now let's say i get a bunch of first_name and last_name combinations and i want to query them in the table, but in combination! It is not allowed either one or the other. In SQL this would look like:

`SELECT * from person WHERE (first_name,last_name) IN (('thomas','mueller'),('robin','hood'));`

So this will definetly take care about all  'thomas mueller' and 'robin hood' records.

In prisma i do this via a raw query. Is there currently a way to achieve the same with the prisma.person.findMany API?

Note: This scenario is not about unique constraints or concating name in views etc in another solution. I want to be able to explicitly focus on combination of values in an IN - Query. Iam facing the issue in much more complicated scenarios and i definetly need this kind of IN query.

Any ideas?

Thanks a lot for your help and best regards
Andreas

**Comment by nurul3101** [comment, marked as answer]
Hello @andreasvh-conceto 👋 
Will a query like this work for your use case?
```typescript
import { Prisma, PrismaClient } from "@prisma/client";

const prisma = new PrismaClient({
  log: ["query", "info", "warn"],
});

// A `main` function so that you can use async/await
async function main() {
  await prisma.person.findMany({
    where: {
      OR: [
        {
          AND: [
            {
              first_name: "thomas",
            },
            {
              last_name: "mueller",
            },
          ],
        },
        {
          AND: [
            {
              first_name: "robin",
            },
            {
              last_name: "hood",
            },
          ],
        },
      ],
    },
  });
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

**Reply by andreasvh-conceto** [reply]
Yes thanks this will work. But in my scenario I get often more than 500-1000 pairs to compare. So the query to create in prisma becomes really big. Do you think there will be a performance issue? I mean concatenation of ofer 1000 and conditions could be really large, but the in condition in my sql would nearly do the same :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
